### PR TITLE
Use GCC static analysis when available

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
> `x86_64-w64-mingw32-gcc: fatal error: Killed signal terminated program cc1`

/me goes to file a GCC bug...

File: meson.build [hunk 1]
```diff
@@ -108,6 +108,9 @@ warning_flags = [
   '-Wvla',
   '-Wwrite-strings'
 ]
+if not meson.is_cross_build()
```
Comment (superm1) [inline meson.build:111]: Can we detect mingw specifically instead?

Comment (hughsie) [reply meson.build:111]: I don't think so, as this happens before `cc = meson.get_compiler('c')`

Comment (superm1) [reply meson.build:111]: How about using
```
if host_machine.system() == 'windows'
```

Comment (hughsie) [reply meson.build]: Lets try that; fixed

**Review by superm1 — APPROVED** [review]

File: meson.build (hunk 1 above)
Comment (superm1) [reply meson.build]: Maybe leave a comment with your GCC bug so future selves can find it when it's fixed to take this out.

Comment (hughsie) [reply meson.build]: It might just be the mingw toolchain; investigating after I've done all my meetings for the day.

**Pull request merged**